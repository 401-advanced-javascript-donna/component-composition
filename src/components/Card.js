import React from 'react';
import PropTypes from 'prop-types';

const Card = ({ _id, name, image, title }) => {
  return (
    <>
      {title && <header><h3>{name}</h3></header>}
      <figure>
        <img src={image} alt={name} />
      </figure>
      <p>{name} has {_id}</p>
    </>
  );
};

Card.propTypes = {
  title: PropTypes.string,
  _id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired
};

export default Card;
