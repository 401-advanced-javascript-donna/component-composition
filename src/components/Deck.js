import React from 'react';
import PropTypes from 'prop-types';

const Deck = ({ children }) => {
  return (
    <section>
      <header>
        <h2>Characters</h2>
      </header>
      {children}
    </section>
  );
};

Deck.propTypes = {
  children: PropTypes.node.isRequired
};

export default Deck;
