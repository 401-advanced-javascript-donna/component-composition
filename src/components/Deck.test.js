import React from 'react';
import { shallow } from 'enzyme';
import Deck from './Deck';
import Card from './Card';
import { characters } from '../content.json';

describe('Deck component', () => {
  const characterNodes = characters.map(item => {
    return <div key={item._id}><Card {...item} /></div>;
  });
  it('renders Deck', () => {
    const wrapper = shallow(<Deck title="characters">{characterNodes}</Deck>);
    expect(wrapper).toMatchSnapshot();
  });
});
