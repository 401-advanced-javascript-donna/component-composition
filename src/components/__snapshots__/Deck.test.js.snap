// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deck component renders Deck 1`] = `
<section>
  <header>
    <h2>
      characters
    </h2>
  </header>
  <div
    key="5da237699734fdcb7bef8f51"
  >
    <Card
      _id="5da237699734fdcb7bef8f51"
      image="https://vignette.wikia.nocookie.net/heyarnold/images/f/f5/Arnold.png/revision/latest?cb=20181027162333"
      name="Arnold Shortman"
    />
  </div>
  <div
    key="5da237699734fdcb7bef8f53"
  >
    <Card
      _id="5da237699734fdcb7bef8f53"
      image="https://vignette.wikia.nocookie.net/heyarnold/images/7/7b/DB87AC2A-C2C2-4DAB-AECF-B9F7273C746B.png/revision/latest/scale-to-width-down/306?cb=20190816101257"
      name="Grandma Gertie"
    />
  </div>
  <div
    key="5da237699734fdcb7bef8f52"
  >
    <Card
      _id="5da237699734fdcb7bef8f52"
      image="https://vignette.wikia.nocookie.net/heyarnold/images/7/79/GrandpaPhil.png/revision/latest/scale-to-width-down/300?cb=20160924030436"
      name="Grandpa Phil"
    />
  </div>
  <div
    key="5da237699734fdcb7bef8f56"
  >
    <Card
      _id="5da237699734fdcb7bef8f56"
      image="https://vignette.wikia.nocookie.net/heyarnold/images/2/2e/Miles.jpg/revision/latest/scale-to-width-down/243?cb=20101115140639"
      name="Miles Shortman"
    />
  </div>
  <div
    key="5da237699734fdcb7bef8f54"
  >
    <Card
      _id="5da237699734fdcb7bef8f54"
      image="https://vignette.wikia.nocookie.net/heyarnold/images/d/d9/Mitzi.png/revision/latest/scale-to-width-down/310?cb=20171212070331"
      name="Mitzi"
    />
  </div>
  <div
    key="5da237699734fdcb7bef8f55"
  >
    <Card
      _id="5da237699734fdcb7bef8f55"
      image="https://vignette.wikia.nocookie.net/heyarnold/images/4/42/Arnie.jpg/revision/latest/scale-to-width-down/310?cb=20110109195825"
      name="Arnie"
    />
  </div>
  <div
    key="5da237699734fdcb7bef8f57"
  >
    <Card
      _id="5da237699734fdcb7bef8f57"
      image="https://vignette.wikia.nocookie.net/heyarnold/images/f/fa/Stella.jpg/revision/latest?cb=20101225143956"
      name="Stella Shortman"
    />
  </div>
  <div
    key="5da237699734fdcb7bef8f59"
  >
    <Card
      _id="5da237699734fdcb7bef8f59"
      image="https://vignette.wikia.nocookie.net/heyarnold/images/f/f7/Grandpa%27s_grandfather.png/revision/latest/scale-to-width-down/310?cb=20190228081015"
      name="Grandpa's grandfather"
    />
  </div>
  <div
    key="5da237699734fdcb7bef8f5a"
  >
    <Card
      _id="5da237699734fdcb7bef8f5a"
      image="https://vignette.wikia.nocookie.net/heyarnold/images/c/c5/Oskar_Kokoshka.png/revision/latest/scale-to-width-down/310?cb=20180707032418"
      name="Oskar Kokoshka"
    />
  </div>
  <div
    key="5da237699734fdcb7bef8f58"
  >
    <Card
      _id="5da237699734fdcb7bef8f58"
      image="https://vignette.wikia.nocookie.net/heyarnold/images/e/e0/Grandpa%27s_father.png/revision/latest/scale-to-width-down/310?cb=20190228081112"
      name="Grandpa's father"
    />
  </div>
  <div
    key="5da237699734fdcb7bef8f5c"
  >
    <Card
      _id="5da237699734fdcb7bef8f5c"
      image="https://vignette.wikia.nocookie.net/heyarnold/images/6/62/Nancy.jpg/revision/latest/scale-to-width-down/310?cb=20110423055923"
      name="Nancy"
    />
  </div>
  <div
    key="5da237699734fdcb7bef8f5b"
  >
    <Card
      _id="5da237699734fdcb7bef8f5b"
      image="https://vignette.wikia.nocookie.net/heyarnold/images/f/fe/Suzie.jpg/revision/latest/scale-to-width-down/310?cb=20110223003447"
      name="Suzie Kokoshka"
    />
  </div>
  <div
    key="5da237699734fdcb7bef8f5e"
  >
    <Card
      _id="5da237699734fdcb7bef8f5e"
      image="https://vignette.wikia.nocookie.net/heyarnold/images/6/60/Mr._Hyunh.png/revision/latest/scale-to-width-down/310?cb=20180707031441"
      name="Mr. Hyunh"
    />
  </div>
  <div
    key="5da237699734fdcb7bef8f61"
  >
    <Card
      _id="5da237699734fdcb7bef8f61"
      image="https://vignette.wikia.nocookie.net/heyarnold/images/e/e8/762C69BE-23BA-48BA-9B14-6AF3246FCD3F.jpeg/revision/latest/scale-to-width-down/304?cb=20190106211607"
      name="Lana Vail"
    />
  </div>
  <div
    key="5da237699734fdcb7bef8f60"
  >
    <Card
      _id="5da237699734fdcb7bef8f60"
      image="https://vignette.wikia.nocookie.net/heyarnold/images/8/80/Mr._Smith.jpg/revision/latest?cb=20110206233402"
      name="Mr. Smith"
    />
  </div>
  <div
    key="5da237699734fdcb7bef8f62"
  >
    <Card
      _id="5da237699734fdcb7bef8f62"
      image="https://vignette.wikia.nocookie.net/heyarnold/images/9/9c/Mr._Purdy_%28character%29.jpeg/revision/latest?cb=20190125214742"
      name="Mr. Purdy (character)"
    />
  </div>
  <div
    key="5da237699734fdcb7bef8f66"
  >
    <Card
      _id="5da237699734fdcb7bef8f66"
      image="https://vignette.wikia.nocookie.net/heyarnold/images/5/5e/Olga_Pataki.png/revision/latest/scale-to-width-down/310?cb=20180124053913"
      name="Olga Pataki"
    />
  </div>
  <div
    key="5da237699734fdcb7bef8f65"
  >
    <Card
      _id="5da237699734fdcb7bef8f65"
      image="https://vignette.wikia.nocookie.net/heyarnold/images/6/6a/Miriam_Pataki.png/revision/latest/scale-to-width-down/310?cb=20170622020421"
      name="Miriam Pataki"
    />
  </div>
  <div
    key="5da237699734fdcb7bef8f6b"
  >
    <Card
      _id="5da237699734fdcb7bef8f6b"
      image="https://vignette.wikia.nocookie.net/heyarnold/images/e/e8/Timberly.png/revision/latest/scale-to-width-down/310?cb=20171210045406"
      name="Timberly Johanssen"
    />
  </div>
  <div
    key="5da237699734fdcb7bef8f67"
  >
    <Card
      _id="5da237699734fdcb7bef8f67"
      image="https://vignette.wikia.nocookie.net/heyarnold/images/9/9c/GeraldShow.png/revision/latest/scale-to-width-down/310?cb=20181027162223"
      name="Gerald Johanssen"
    />
  </div>
</section>
`;
